src/java.base/share/classes/java/lang/Runtime.java

```diff
@@ -816,12 +816,14 @@ void loadLibrary0(Class<?> fromClass, String libname) {
      * @since  9
      */
     public static Version version() {
-        if (version == null) {
-            version = new Version(VersionProps.versionNumbers(),
+        var v = version;
+        if (v == null) {
+            v = new Version(VersionProps.versionNumbers(),
                     VersionProps.pre(), VersionProps.build(),
                     VersionProps.optional());
+            version = v;
```

Comment:
Can't this just be `return version = new Version(...` than reassigning to a local variable for no good?

Reply:
It's matter of style. I've seen both styles are acceptable in JDK codebase. I personally prefer placing assigning each variable into separate lines.

Reply:
Doesn't using `return version = new Version(...` allow one local variable to be effectively final and save two aload/astore opcodes? Guess jvm optimizes it